src/backend/luna.rs

```diff
@@ -0,0 +1,100 @@
+use rusb::{
+    GlobalContext, DeviceHandle,
+};
+use std::collections::VecDeque;
+use std::thread;
+use std::sync::mpsc::{channel, Receiver};
+use std::time::Duration;
+
+const VID: u16 = 0x1d50;
+const PID: u16 = 0x615b;
+
+const ENDPOINT: u8 = 0x81;
+
+const READ_LEN: usize = 0x4000;
+
+pub struct LunaDevice {
+    handle: Option<DeviceHandle<GlobalContext>>,
+    worker: Option<thread::JoinHandle<DeviceHandle<GlobalContext>>>,
+    receiver: Option<Receiver<Vec<u8>>>,
+}
+
+impl LunaDevice {
+    pub fn open() -> Self {
+        let handle = rusb::open_device_with_vid_pid(VID, PID).unwrap();
```

Comment:
I'd rather not see `unwrap` inside the implementation here - let's define an error type and pass it back to `main`.